relacy/relacy_std.hpp

```diff
@@ -19,64 +19,78 @@
-//    using rl::atomic_int_least16_t;
-//    using rl::atomic_uint_least16_t;
-//    using rl::atomic_int_least32_t;
-//    using rl::atomic_uint_least32_t;
-//    using rl::atomic_int_least64_t;
-//    using rl::atomic_uint_least64_t;
-//    using rl::atomic_int_fast8_t;
-//    using rl::atomic_uint_fast8_t;
-//    using rl::atomic_int_fast16_t;
-//    using rl::atomic_uint_fast16_t;
-//    using rl::atomic_int_fast32_t;
-//    using rl::atomic_uint_fast32_t;
-//    using rl::atomic_int_fast64_t;
-//    using rl::atomic_uint_fast64_t;
-    using rl::atomic_intptr_t;
-    using rl::atomic_uintptr_t;
-    using rl::atomic_size_t;
-//    using rl::atomic_ssize_t;
-    using rl::atomic_ptrdiff_t;
-//    using rl::atomic_intmax_t;
-//    using rl::atomic_uintmax_t;
+#    define atomic_char rlimp::atomic_char
+#    define atomic_schar rlimp::atomic_schar
+#    define atomic_uchar rlimp::atomic_uchar
+#    define atomic_short rlimp::atomic_short
+#    define atomic_ushort rlimp::atomic_ushort
+#    define atomic_int rlimp::atomic_int
+#    define atomic_uint rlimp::atomic_uint
+#    define atomic_long rlimp::atomic_long
+#    define atomic_ulong rlimp::atomic_ulong
+#    define atomic_llong rlimp::atomic_llong
+#    define atomic_ullong rlimp::atomic_ullong
+//#    define atomic_char16_t rlimp::atomic_char16_t
+//#    define atomic_char32_t rlimp::atomic_char32_t
+#    define atomic_wchar_t rlimp::atomic_wchar_t
 
-    using rl::mutex;
-    using rl::recursive_mutex;
-    using rl::condition_variable;
-    using rl::condition_variable_any;
+//#    define atomic_int_least8_t rlimp::atomic_int_least8_t
+//#    define atomic_uint_least8_t rlimp::atomic_uint_least8_t
+//#    define atomic_int_least16_t rlimp::atomic_int_least16_t
+//#    define atomic_uint_least16_t rlimp::atomic_uint_least16_t
+//#    define atomic_int_least32_t rlimp::atomic_int_least32_t
+//#    define atomic_uint_least32_t rlimp::atomic_uint_least32_t
+//#    define atomic_int_least64_t rlimp::atomic_int_least64_t
+//#    define atomic_uint_least64_t rlimp::atomic_uint_least64_t
+//#    define atomic_int_fast8_t rlimp::atomic_int_fast8_t
+//#    define atomic_uint_fast8_t rlimp::atomic_uint_fast8_t
+//#    define atomic_int_fast16_t rlimp::atomic_int_fast16_t
+//#    define atomic_uint_fast16_t rlimp::atomic_uint_fast16_t
+//#    define atomic_int_fast32_t rlimp::atomic_int_fast32_t
+//#    define atomic_uint_fast32_t rlimp::atomic_uint_fast32_t
+//#    define atomic_int_fast64_t rlimp::atomic_int_fast64_t
+//#    define atomic_uint_fast64_t rlimp::atomic_uint_fast64_t
+#    define atomic_intptr_t rlimp::atomic_intptr_t
+#    define atomic_uintptr_t rlimp::atomic_uintptr_t
+#    define atomic_size_t rlimp::atomic_size_t
+//#    define atomic_ssize_t rlimp::atomic_ssize_t
+#    define atomic_ptrdiff_t rlimp::atomic_ptrdiff_t
+//#    define atomic_intmax_t rlimp::atomic_intmax_t
+//#    define atomic_uintmax_t mutex rlimp::atomic_uintmax_t
+
+#    define mutex rlimp::mutex

```

Comment:
Does mutex lead to the same problems?
It's pretty common name, so redefining it is unfortunate...